monai/handlers/wandb_handlers.py

```diff
@@ -0,0 +1,202 @@
+                when every iteration completed. The default behavior is to print loss from output[0] as
+                output is a decollated list and we replicated loss value for every item of the decollated
+                list. `engine.state` and `output_transform` inherit from the
+                ignite concept: https://pytorch.org/ignite/concepts.html#state, explanation and usage
+                example are in the tutorial:
+                https://github.com/Project-MONAI/tutorials/blob/master/modules/batch_output_transform.ipynb.
+            global_epoch_transform: A callable that is used to customize global epoch number. For example,
+                in evaluation, the evaluator engine might want to use trainer engines epoch number when
+                plotting epoch vs metric curves.
+            state_attributes: Expected attributes from `engine.state`, if provided, will extract them when
+                epoch completed.
+            tag_name: When iteration output is a scalar, tag_name is used to plot, defaults to `'Loss'`.
+        """
+        if wandb.run is None:
+            raise wandb.Error("You must call `wandb.init()` before using `WandbStatsHandler()`")
+
+        self.iteration_log = iteration_log
+        self.epoch_log = epoch_log
+        self.epoch_event_writer = epoch_event_writer
+        self.epoch_interval = epoch_interval
+        self.iteration_event_writer = iteration_event_writer
+        self.iteration_interval = iteration_interval
+        self.output_transform = output_transform
+        self.global_epoch_transform = global_epoch_transform
+        self.state_attributes = state_attributes
+        self.tag_name = tag_name
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Register a set of Ignite Event-Handlers to a specified Ignite engine.
+
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        if self.iteration_log and not engine.has_event_handler(self.iteration_completed, Events.ITERATION_COMPLETED):
+            engine.add_event_handler(
+                Events.ITERATION_COMPLETED(every=self.iteration_interval), self.iteration_completed
+            )
+        if self.epoch_log and not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
+            engine.add_event_handler(Events.EPOCH_COMPLETED(every=self.epoch_interval), self.epoch_completed)
+
+    def epoch_completed(self, engine: Engine) -> None:
+        """
+        Handler for train or validation/evaluation epoch completed Event. Write epoch level events
+        to Weights & Biases, default values are from Ignite `engine.state.metrics` dict.
+
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        if self.epoch_event_writer is not None:
+            self.epoch_event_writer(engine)
```

Comment:
In the doc-string, it says that the `epoch_event_writer: ..... Must accept the parameter "engine" and "summary_writer" ....`. Whereas, I don't see the `summary_writer` parameter here. It's a little bit confusing for me. Could you please explain it?

Thanks,
Bin